packages/genesys-spark-components/src/utils/date/temporal.ts

```diff
@@ -23,6 +23,10 @@ export function formatPlainDate(
   // would happen if we passed the values in via the Date constructor, where
   // they are interpreted as UTC
   formatDate.setFullYear(date.year);
+
+  // Set date to 1 first to avoid rollover issues when changing month
+  formatDate.setDate(1);
+
   formatDate.setMonth(date.month - 1);
```

Comment:
I think this can be `formatDate.setMonth(date.month - 1, 1);` and you don't need the new `setDate` above